src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -65,6 +66,26 @@ def __init__(self, schedule=None):
         # Reference to Schedule to which this symbol table belongs.
         self._schedule = schedule
 
+    def new_symbol_name(self, root_name=None):
+        '''Create a symbol name that is not in the symbol table. If the
+        `root_name` argument is not supplied then the name is
+        generated internally, otherwise the `root_name` is used. If
+        required, an additional integer is appended to avoid clashes.
+
+        :param str root_name: the name to use when creating a new \
+        symbol name. This will be appended with an integer if the name \
+        clashes with an existing symbol name.
+
+        '''
+        if not root_name:
```

Comment:
Are you aware that `st.new_symbol_name("")` will make this condition True? This is probably a good think as languages do not allow variables that start with a numerical digit , but I think it should be documented in the method docstring. And tested.

Reply:
Also add a type check to the root_name parameter to make sure that a string is passed.

Reply:
Good points. Done.